modular_splurt/code/modules/arousal/arousal.dm

```diff
@@ -2,20 +2,44 @@
 	. = ..()
 	L.receive_climax(src, Lgen, G, spillage, forced = forced)
 
-/mob/living/receive_climax(mob/living/partner, obj/item/organ/genital/receiver, obj/item/organ/genital/source_gen, spill, forced)
-	. = ..()
+/mob/living/proc/receive_climax(mob/living/partner, obj/item/organ/genital/receiver, obj/item/organ/genital/source, spill, forced)
+	//gregnancy...
+	if(!spill && istype(source, /obj/item/organ/genital/penis) && \
+		istype(receiver, /obj/item/organ/genital/vagina) && getorganslot(ORGAN_SLOT_WOMB))
+		var/obj/item/organ/genital/penis/peenus = source
+		if(!(locate(/obj/item/genital_equipment/condom) in peenus.contents))
+			impregnate(partner, getorganslot(ORGAN_SLOT_WOMB), src.type)
 
 	if(!receiver || spill || forced)
 		return
 
-	receiver.climax_modify_size(partner, source_gen)
+	receiver.climax_modify_size(partner, source)
+
+//handles impregnation, also prefs
+/mob/living/proc/impregnate(mob/living/partner, obj/item/organ/W, baby_type)
+	if(!W)
+		return
+	var/can_impregnate = 100
+	if(partner?.client?.prefs)
+		can_impregnate = partner.client.prefs.virility
+	var/can_get_pregnant = (client?.prefs?.fertility && !is_type_in_typecache(src.type, GLOB.pregnancy_blocked_mob_typecache))
+	if(!(can_impregnate || can_get_pregnant))
```

Comment:
```suggestion
	if(!(can_impregnate && can_get_pregnant))
```
Should probably be something like this or else it'll just ignore the prefs of a parent with the mechanics deactivated if the other one does have them activated or has no client at all